tests/test_device.py

```diff
@@ -694,3 +708,129 @@ def test_outdated_API(self, monkeypatch):
             m.setattr(qml, "version", lambda: "0.0.1")
             with pytest.raises(DeviceError, match="plugin requires PennyLane versions"):
                 qml.device("default.qubit", wires=0)
+
+
+class TestBatchExecution:
+    """Tests for the batch_execute method."""
+
+    def test_results_no_interface(self, tol):
+        """Tests that the correct results are computed using tapes without an interface."""
+        # skip test if interface cannot be imported
+
+        qml.enable_tape()
+
+        a = 0.1
+        b = 0.2
+        tapes = (tp1, tp2)
+
+        dev = qml.device('default.qubit', wires=2)
+        res = dev.batch_execute(tapes)
+
+        exp1 = [np.cos(a), -np.cos(a) * np.sin(b)]
+        exp2 = np.cos(a)
+        expected = [exp1, exp2]
+
+        # need to compare element-wise because np.allclose
+        # does not work for a mix of lists and tensors
+        for r, e in zip(res, expected):
+            assert np.allclose(r, e, atol=tol, rtol=0)
+        assert isinstance(res, list)
+
+        qml.disable_tape()
```

Comment:
Just something that comes to mind: if any part of the above test fails, then `qml.disable_tape()` won't be called, causing subsequent tests to continue being in tape mode (even if they shouldn't be).

We can make `enable_tape()` and `disable_tape()` part of the test set up and tear down to avoid this:


```python
@pytest.fixture()
def tape_mode():
    qml.enable_tape()
    yield
    qml.disable_tape()

def test_something(tape_mode):
```

Reply:
Or, to automatically use the setup/teardown on a class of tests: https://docs.pytest.org/en/latest/fixture.html#autouse-fixtures-xunit-setup-on-steroids

Reply:
Good idea! Although, as you point out, I don't need it here...